### DISCUSSION TITLE
Organize company customers

### DISCUSSION THREAD

**Discussion by WeAreHaus** [vendurehq/vendure#2194] [Q&A] [answered]
Hi!

We are evaluating Vendure for an upcoming B2B e-commerce project. Our impression of the platform is very good and we like the architecture, the powerful customization possibilities and the great documentation. There are still some aspects to work out before we can make a final decision and I'm hoping you can help us out.

Our customer is a wholesale reseller to multiple hotel chains and the end users are employees working at the hotels.

In Vendure, a customer represent an individual but we would like it to represent a company. A feature that Vendure has out-of-the-box is multiple shipping addresses per customer, which is a requirement. However, Vendure treats customers as users, which makes it difficult to use the Customer entity to represent a hotel since each hotel can have multiple users.

We would like to organize the customers/users as: company - business unit - user. And then be able to associate multiple addresses with one company.

How can this be achieved by using as much standard functionality as possible?

Thank you in advance!

**Comment by michaelbromley** [comment, marked as answer]
Hi,

This is possible by means of a custom plugin in which you define new entities to represent Company & BusinessUnit. You can then define the appropriate relations, e.g. Company -> many BusinessUnits -> many Customers.

A close partner of ours that specializes in B2B is actually developing a B2B suite of plugins which includes modelling this exact scenario, and in the future we plan on making this available in our upcoming plugin marketplace (no fixed date yet though I'm afraid). If you join our Discord server (https://vendure.io/community) you may be able to find help from some of the devs who have been working on this.

**Reply by WeAreHaus** [reply]
Excellent, thank you very much for the swift reply!
